app/components/autocomplete/slash_suggestion/slash_suggestion.js

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {
+    FlatList
+} from 'react-native';
+
+import {RequestStatus} from 'mattermost-redux/constants';
+
+import {makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import SlashSuggestionItem from './slash_suggestion_item';
+
+const SLASH_REGEX = /(^\/)([a-zA-Z-]*)$/;
+const FIVE_MINS = 1000 * 60 * 5;
+
+export default class SlashSuggestion extends Component {
+    static propTypes = {
+        actions: PropTypes.shape({
+            getAutocompleteCommands: PropTypes.func.isRequired
+        }).isRequired,
+        currentTeamId: PropTypes.string.isRequired,
+        commands: PropTypes.array,
+        commandsRequest: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        onChangeText: PropTypes.func.isRequired,
+        onResultCountChange: PropTypes.func.isRequired,
+        value: PropTypes.string
+    };
+
+    static defaultProps = {
+        defaultChannel: {},
+        value: ''
+    };
+
+    state = {
+        active: false,
+        suggestionComplete: false,
+        dataSource: [],
+        lastCommandRequest: 0
+    };
+
+    componentWillReceiveProps(nextProps) {
+        const {currentTeamId} = this.props;
+        const {
+            commands: nextCommands,
+            commandsRequest: nextCommandsRequest,
+            currentTeamId: nextTeamId,
+            value: nextValue
+        } = nextProps;
+
+        if (currentTeamId !== nextTeamId) {
+            this.setState({
+                lastCommandRequest: 0
+            });
+        }
+
+        const regex = SLASH_REGEX;
```

Comment:
We could skip the variable assignment here.

Reply:
Will skip.